### DISCUSSION TITLE
Add Ping Button for Web Sockets

### DISCUSSION THREAD

**Discussion by komape** [Kong/insomnia#5610] [Ideas]
One of the control frames for web sockets are ping and pong messages as described in the [mdn web docs](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_servers#pings_and_pongs_the_heartbeat_of_websockets) and in [section 5.5.2 and 5.5.3 of RFC 6455](https://datatracker.ietf.org/doc/html/rfc6455#section-5.5.2). Closing the socket also uses a control frame which is sent if you click on the `Disconnect` button in Insomnia. Would be great, if there is also a `Ping` button to check the connection to the server. What do you think of it?

**Comment by rishabhgup-adb** [comment]
I was thinking of the same and happy to take up the work for enhancing Ping/Pong from websocket. Features I think insomnia needs:
1. Ability to send Ping Frame
2. View Ping/Pong frame received in the Req/Res window
